### PR TITLE
Tidy containers

### PR DESCRIPTION
**Pull request by thewilkybarkid**
Follow up to (and so currently includes) #892.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -3,5 +3,6 @@ IMAGE_TAG=develop
 ENVIRONMENT=production
 DESCRIPTION=
 TEST_HTML=
+NODE_VERSION=6.14.3
```
Comment (davidcmoulton) [inline .env]: This makes my eyes bleed. Update of this to LTS coming soon (as a separate PR).

Comment (giorgiosironi) [reply .env]: Expires in 2019 fortunately

File: Dockerfile.assets [hunk 2]
```diff
@@ -1,12 +1,12 @@
 ARG image_tag=latest
 FROM elifesciences/pattern-library_assets-builder:${image_tag} AS assets-builder
-
 FROM tianon/true:latest
-ENV PROJECT_FOLDER=/srv/pattern-library
```
Comment (giorgiosironi) [inline Dockerfile.assets]: `PROJECT_FOLDER` is sometimes read by pipelines to determine where to find the smoke tests. This image should not be affected https://github.com/elifesciences/elife-jenkins-workflow-libs/search?q=PROJECT_FOLDER&unscoped_q=PROJECT_FOLDER

File: Dockerfile.assets-builder [hunk 3]
```diff
@@ -1,59 +1,27 @@
...
-# end base image
-
-# potentially base image
-RUN npm install -g gulp-cli
-# end potentially base image
-
-RUN apt-get update && apt-get install -y \
-    bzip2 \
-    g++ \
-    make
```
Comment (giorgiosironi) [inline Dockerfile.assets-builder]: Where have `bzip2`, `g++` and `make` gone? Not necessary anymore, what they were used for?

File: Dockerfile.assets-builder [hunk 4]
```diff
@@ -1,59 +1,27 @@
...
-COPY --chown=elife:elife assets/fonts/ ${PROJECT_FOLDER}/assets/fonts
-RUN gulp --environment ${environment} fonts
-COPY --chown=elife:elife assets/img/ ${PROJECT_FOLDER}/assets/img
-RUN gulp --environment ${environment} img
+COPY assets/fonts/ assets/fonts/
+RUN node_modules/.bin/gulp --environment ${environment} fonts
 
-COPY --chown=elife:elife assets/js/ ${PROJECT_FOLDER}/assets/js
-RUN gulp --environment ${environment} js
+COPY assets/img/ assets/img/
```
Comment (giorgiosironi) [inline Dockerfile.assets-builder]: Containers running as `root` in general should be avoided to give them the least privilege. This is a build phase so may be different. Then the `node` image uses the `root` user as default :thinking:

File: docker-compose.override.yml [hunk 5]
```diff
@@ -1,13 +1,19 @@
...
+            - ./package.json:/package.json
+            - node_modules:/node_modules
     assets-builder:
         volumes:
             - ./:/srv/pattern-library
             - /srv/pattern-library/node_modules
             - source_assets:/srv/pattern-library/source/assets
             - ./.container_source_assets:/srv/pattern-library/.container_source_assets
             - test_build:/srv/pattern-library/test/build
+            - node_modules:/node_modules
```
Comment (giorgiosironi) [inline docker-compose.override.yml]: there are two `node_modules` folders in this list now?

File: Dockerfile.assets-builder (hunk 3 above)
Comment (thewilkybarkid) [reply Dockerfile.assets-builder]: The `node` image extends `buildpack-deps` which already contains them.

File: Dockerfile.assets-builder (hunk 4 above)
Comment (thewilkybarkid) [reply Dockerfile.assets-builder]: Done in 769c7aa

**Comment by thewilkybarkid** [comment]
@davidcmoulton Should be fixed.

**Comment by thewilkybarkid** [comment]
The slow behaviour was happening to me before this change, are you only now seeing it?

**Comment by davidcmoulton** [comment]
It seems worse now. If you've seen it before though then doesn't seem like it's introduced by this change, so can ignore for this PR.

**Pull request merged**